application.py

```diff
@@ -56,28 +67,64 @@ def auth():
     password = request.form.get("password")
     queryres = userdata.query.filter_by(username = uname).first()
     if queryres is not None:
-        print(password == queryres.password)
-        print(uname == queryres.password)
         if password == queryres.password and uname == queryres.username:
             session["user"] = uname
             return redirect(url_for("userhome"))
         else :
             flash("The entered passowrd of the account doesnot match.")
-            return render_template("register.html")
+            return redirect(url_for("register"))
     else:
         flash("Your logging credentials are invalid.")
-        return render_template("register.html")
+        return redirect(url_for("register"))
 
 @app.route("/logout")
 def logout():
     session.pop("user",None)
-    return render_template("register.html")
+    flash("You are logout.")
+    return redirect(url_for("register"))
 
 @app.route("/userhome")
 def userhome():
     if "user" in session:
         user = session["user"]
-        return render_template("home.html")
+        booksn = books.query.all()
+        return render_template("home.html", books = booksn )
     else:
         flash("Login with the credentails.")
-        return render_template("register.html")
\ No newline at end of file
+        return redirect(url_for("register"))
+
+@app.route("/books/<string:ISBN_number>")
+def bookspage(ISBN_number):
+    pass
+
+@app.route("/search", methods = ["POST"])
+def search():
+    tag=request.form.get("search")
+    tag = string.capwords(tag)
+    # print("tag " + tag)
+    if tag != "":
+        searchque="%{}%".format(tag)
+        # print(searchque)
+        ISBN= books.query.filter(books.ISBN_number.like(searchque)).all()
+        print("ISBN")
+        print(ISBN)
+        if len(ISBN) == 0:
+            usname = books.query.filter(books.name.like(searchque)).all()
+            if len(usname) == 0:
+                print("usname")
+                print(usname)
+                authorw = books.query.filter(books.author.like(searchque)).all()
+                if len(authorw) == 0:
+                    print("author")
+                    print(authorw)
+                    flash("The searched book does not exists.")
```

Comment:
Can you think of simplifying this logic?
Hint : See about " or_ " in sqlalchemy